### PR TITLE
Fix comment editing, add urls for communities

### PR DESCRIPTION
**Pull request by konrad-paszek**
Fixes according to https://app.clickup.com/t/8697w013u & https://app.clickup.com/t/8697w00vy

## Summary by Sourcery

This pull request introduces comment editing functionality and enhances community links throughout the application. It enables users to modify their existing comments and provides direct navigation to community details from various post listings.

New Features:
- Adds the ability for users to edit their comments.

Enhancements:
- Updates the post header and user post list templates to link to community detail pages.

### CODE REVIEW

File: core/tests/test_views.py [hunk 1]
```diff
@@ -679,3 +679,22 @@ def test_fetch_user_comments_with_empty_filter(client: Client, user: User, comme
     response = client.get(f"{reverse('user_comments')}?filter=")
     assert response.status_code == 200
     assert len(response.context_data["comments"]) == 1
+
+
+def test_edit_comment_valid(client: Client, user: User, post: Post) -> None:
+    data = {
```
Comment (jacoor) [inline core/tests/test_views.py]: Thank you for this test, this is very good excercise for test optimisation. 

Hmm this test is kinda weird. Let me describe what I see it does
1. It creates a comment with data from line 685
2. Verifies the comment was created (200) but does not prove the data in the post is what we send, or the parent is correct. 
3. edits the comments
4. Verifies it's ok by checking 302 http code, which is not 200 OK - error page could also be moved. 
5. Refreshes post from DB but does not to anything with that post.
6. Verifies the content after edition is TEST.

THe name of the test is`test_edit_comment_valid` and it creates comment, checks comment creation and later edition.  It does much more than expected. 

Please edit the test to only check editing. Use fixture for post and comment, if not already created. 
The test should follow structure
Given, when, then. The current version is like: 
Given - user. data.
when -> create comment
then -> comment created
When(?) edit response
Then.... 

what I would do is
- Given: use fixture for post and first comment
- When: edit the comment by HTTP request
- Then: verify the comment by retriebving it from database - the call in 699 to post details is not needed. You already have this data in DB and in edit response context. 
The provided steps will make test much faster by using only one HTTP call instead of 3.

**Review by jacoor — CHANGES_REQUESTED** [review]
Very good. The test need optimisation.

File: core/tests/test_views.py [hunk 2]
```diff
@@ -679,3 +679,22 @@ def test_fetch_user_comments_with_empty_filter(client: Client, user: User, comme
     response = client.get(f"{reverse('user_comments')}?filter=")
     assert response.status_code == 200
     assert len(response.context_data["comments"]) == 1
+
+
+def test_edit_comment_valid(client: Client, user: User, post: Post) -> None:
```
Comment (jacoor) [inline core/tests/test_views.py]: you could also add one test for trying to save empty content.

File: core/tests/test_views.py [hunk 3]
```diff
@@ -679,3 +686,19 @@ def test_fetch_user_comments_with_empty_filter(client: Client, user: User, comme
     response = client.get(f"{reverse('user_comments')}?filter=")
     assert response.status_code == 200
     assert len(response.context_data["comments"]) == 1
+
+
+def test_edit_comment_valid(client: Client, user: User, post: Post, comment: Post) -> None:
+    client.force_login(user)
+    edit_response = client.post(reverse("edit_comment", kwargs={"pk": comment.pk}), data={"content": "TEST"})
+    assert edit_response.status_code == 302
+    post.refresh_from_db()
```
Comment (jacoor) [inline core/tests/test_views.py]: Not needed. You are checking HTTP response not DB response. This is not used.

**Review by jacoor — APPROVED** [review]

**Review by jacoor — CHANGES_REQUESTED** [review]
This close

**Review by jacoor — APPROVED** [review]
